lib/devices.go

```diff
@@ -64,18 +64,34 @@ func Stop() {
 // It takes a callback function to be run anytime there is a change in Devices
 func Init(onUpdateCallback func([]Device)) []Device {
 	pollingStop = make(chan bool)
-	onUpdateCallback(Refresh())
 	go func() {
+		logtime, initialDevices := Refresh()
+
+		for initialDevices == nil {
+			time.Sleep(1 * time.Second)
+			addErrorLog("Failed to enumerate devices. Retrying...", logtime, StateError)
+			onUpdateCallback(nil)
+			logtime, initialDevices = Refresh()
+		}
+
+		onUpdateCallback(initialDevices)
+
 		ticker := time.NewTicker(250 * time.Millisecond)
 		defer ticker.Stop()
 
 		for {
 			select {
 			case <-ticker.C:
 				logtime, newDevices := getDevices()
-				changed, mergedDevices := deviceDiff(newDevices, logtime)
-				if changed {
-					onUpdateCallback(mergedDevices)
+
+				if newDevices != nil {
+					changed, mergedDevices := deviceDiff(newDevices, logtime)
+					if changed {
+						onUpdateCallback(mergedDevices)
+					}
+				} else {
+					addErrorLog("Failed to enumerate devices. Retrying...", logtime, StateError)
```

Comment:
`Refresh()` already logs all error cases, does it not? Why create yet another log entry here?